packages/components/src/form-file-upload/index.tsx

```diff
@@ -40,13 +40,17 @@ export function FormFileUpload( {
 		ref.current?.click();
 	};
 
-	const ui = render ? (
-		render( { openFileDialog } )
-	) : (
-		<Button onClick={ openFileDialog } { ...props }>
-			{ children }
-		</Button>
-	);
+	let ui = null;
+	if ( render ) {
+		render( { openFileDialog } );
+	} else {
+		ui = (
+			<Button onClick={ openFileDialog } { ...props }>
+				{ children }
+			</Button>
+		);
+	}
+
```

Comment:
I believe that the correct fix here it is to change the return type of `render` to `JSX.Element`, as the result or `render` are supposed to be assigned to `ui` — this can be verified in the "With Custom Render" Storybook story for the `FormFileUpload` component, which on this PR is broken:

| trunk | This PR |
|---|---|
| <img width="967" alt="Screenshot 2022-10-25 at 16 16 39" src="https://user-images.githubusercontent.com/1083581/197798276-0676d525-eeb7-40a9-a3b8-6f7cd19cafe9.png"> | <img width="971" alt="Screenshot 2022-10-25 at 16 17 16" src="https://user-images.githubusercontent.com/1083581/197798455-86aa16b0-3a07-497e-a2fd-0c7bae74f792.png"> |

Reply:
Fixed! The `render` prop documentation was wrong, saying that "the component does not render any UI" if the prop is passed, so I fixed the docs, too. I understood the void-returning `render` as a function that renders a UI completely outside React, with vanilla JS, by opening the native modal etc.